jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -474,6 +487,15 @@ public Field run() {
 @CallerSensitive
 public static Class<?> forName(String className, boolean initializeBoolean, ClassLoader classLoader)
 	throws ClassNotFoundException
+{
+	return forName(className, initializeBoolean, classLoader, null);
+}
+
+/*[IF JAVA_SPEC_VERSION >= 18]*/
+@CallerSensitiveAdapter
+/*[ENDIF] JAVA_SPEC_VERSION >= 18 */
+private static Class<?> forName(String className, boolean initializeBoolean, ClassLoader classLoader, Class<?> caller)
+	throws ClassNotFoundException
```

Comment:
```suggestion
/*[IF JAVA_SPEC_VERSION >= 18]*/
{
	return forName(className, initializeBoolean, classLoader, null);
}

@CallerSensitiveAdapter
private static Class<?> forName(String className, boolean initializeBoolean, ClassLoader classLoader, Class<?> caller)
	throws ClassNotFoundException
/*[ENDIF] JAVA_SPEC_VERSION >= 18 */
```
This keeps pre-JDK18 code intact.
Similar comments for other private method additions.

Reply:
new changes keep pre-jdk18 code intact.